homeassistant/components/fritzbox_callmonitor/sensor.py

```diff
@@ -117,7 +118,7 @@ async def async_setup_entry(
     )
 
     hass.bus.async_listen_once(
-        EVENT_HOMEASSISTANT_STOP, sensor.async_will_remove_from_hass()
+        EVENT_HOMEASSISTANT_STOP, sensor.async_will_remove_from_hass
```

Comment:
The listener should accept an `Event` parameter. We shouldn't access entity methods from outside the entity.

Extract a helper method from `async_will_remove_from_hass` with the current body of that method inside the entity and set up the listener in `async_added_to_hass` that uses the new helper method as target.